src/main/proto/com/google/api/codegen/config.proto

```diff
@@ -292,6 +300,84 @@ message MethodConfigProto {
 
 }
 
+// `CodeSampleValues` defines sets of parameter values to be used in
+// generating samples.
+message CodeSampleValues {
+  // An identifier, unique to this method, for this set of values.
+  string id = 1;
+
+  // A short, user-visible name for this set of values. This name may
+  // be used in a UI menu, for example.
+  string title = 2;
+
+  // A longer description for this set of values. This name may be
+  // used in a UI hoverbox, for example.
+  string description = 3;
+
+  // A set of parameter values to be used together in samples. Each
+  // member of `parameters` can be specified in one of two formats:
+  //
+  // * `DOTTED.RESOURCE.NAME=VALUE`, to specify the entire value of
+  //   `NAME`.  If `VALUE` is not specified, it is treated as the
+  //    default zero value for `NAME`'s type.
+  // *  `DOTTED.RESOURCE.NAME#FIELD=VALUE`, if `NAME` is required to be
+  //    a specially formatted string using `FIELD` as one of possibly
+  //    several placeholders.
+  repeated string parameters = 4;
+}
+
+// `SampleConfigurations` defines combinations of "molds" (sample code
+// structures as output by the client library generator) and
+// `CodeSampleValues` that should be combined to generate samples.
+message SampleConfiguration {
+  // `Value` references one or more `CodeSampleValues` and provides a
+  // hint as to where it should be presented.
+  message Value{
+    // One or more of the `CodeSampleValues.id`s defined for this
+    // method. If more than one is specified and `explorer_sample` is
+    // set, the hint for relative order of the samples generated is
+    // taken to be the order of this list. If not specified, `ids`
+    // defaults to a list of all defined values for
+    // `MethodConfigProto.code_sample_values.id`. If any element of
+    // `ids` is invalid (i.e., is not found in
+    // `MethodConfigProto.code_sample_values.id`) and `delete` is not
+    // set, then an fatal error is raised at code-generation time.
+    repeated string ids = 1;
+
+    // Whether the samples specified by this configuration should also
+    // be used as configurable "Explorer Samples" suitable for
+    // inclusion in a UI element.
+    bool explorer_sample = 2;
+
+    // Whether the samples specified by this configuration should also
+    // be used as "Customary Samples" that are included within code
+    // code comments in the surface of each generated client library.
+    bool customary_sample = 3;
```

Comment:
"customary" doesn't really have any meaning out of context. Maybe "surface_sample" would be better, or alternatively "in_code_sample".

Reply:
Fair enough. Going for "in_code_sample" for clarity and specificity.